widgetsnbextension/setup.py

```diff
@@ -19,26 +16,19 @@
 IS_REPO = (HERE.parent / '.git').exists()
 JS_DIR = HERE / 'src'
 
-# Representative files that should exist after a successful build
-js_targets = [
-    HERE / 'widgetsnbextension' / 'static' / 'extension.js'
-]
-
 data_files_spec = [(
     'share/jupyter/nbextensions/jupyter-js-widgets', 'widgetsnbextension/static', 'extension.js*'),
     ('etc/jupyter/nbconfig/notebook.d' , '.', 'widgetsnbextension.json')
 ]
 
-cmdclass = create_cmdclass('jsdeps', data_files_spec=data_files_spec)
-js_command = combine_commands(
-    install_npm(JS_DIR, npm=['yarn'], build_cmd='build'),
-    ensure_targets(js_targets),
+post_develop = npm_builder(
+    build_cmd="yarn", source_dir="src", build_dir=JS_DIR
 )
 
-if IS_REPO:
-    cmdclass["jsdeps"] = js_command
-else:
-    cmdclass["jsdeps"] = skip_if_exists(js_targets, js_command)
+cmdclass = wrap_installers(post_develop=post_develop)
 
 if __name__ == '__main__':
-    setup(cmdclass=cmdclass)
+    setup(
+        cmdclass=cmdclass,
+        data_files=get_data_files(data_files_spec),
```

Comment:
@jtpio - I've been wondering about how jupyter_packaging handles data files that don't exist until after the build process. Part of the reason for the complicated build process before was that the data files could not be calculated until after the js was built. How are the data files retrieved with `get_data_files` before they are built now?